internal/vds/vds.cpp

```diff
@@ -29,7 +29,7 @@ void vdsbuffer_delete(struct vdsbuffer* buf) {
     *buf = vdsbuffer {};
 }
 
-int axis_todim(axis ax) {
+int axis_todim(api_axis_name ax) {
```

Comment:
Not a fan of the name `api_axis_name` tbh. The `api` part doesn't add anything for me. Maybe if we prefix everything in the C API with `api` it would make sense [1]. But as a lone prefix it's just another meaningless term. After all it doesn't specify what api it refers to. I suggest shortening it too `axis_name`

[1] But then perhaps `api_`is not the best prefix

Reply:
I am more fine with it.
I think the point was to clearly distinguish between things in API and things in openvds.
If we do not have openvds prefix, we need to have an api prefix.
So it is either api, or vdsslice (I've already been through all 5 stages of grief and accepted the name).

I think it should make it either to not mix with axis names from openvds.

Reply:
I think that it is useful to have some kind of indication that this naming of the axis is from our side (prefix `api`, `vdsslice`, `openvds_adapter`, `request`...). As we have (at least) two notions of axes: the axes defined within (Open)VDS and in our requests/API. There are quite some cases where the axis do not have the same name, but there has to be a mapping between the axis specified in a request and axes in the VDS dataset that is currently used. 

In the later pull requests, I introduce an axis class that has members to return names/indices for the API/vds-slice (referred to as `api`) naming convention and the OpenVDS (referred to as `vds`) convention. I see that `api` is ambiguous, but it was a tempting term as it is short. 

Would a more expressive name be helpful? Maybe `request_axis_name`?

Reply:
We either introduce a consistent naming scheme for referencing our API, or we don't. My point is, this introduces a new term for referencing our C API, but only for a single type. And in the follow-up PR `api` is used once more, but not in the C API that time, then it's on `Axis`.  So it just another noisy word that thrown around. It adds nothing to my understanding, but bloats all code it touches. 

And API is poor name too. (naming is hard, so no judgement). whenever I leave this code for a bit and come back I always have to think: api, which api? oh _our_ api, not openvds's api.

And keeping in mind that in the follow-up PR's we aim to 1) abstract aways OpenVDS into 2 classes and 2) namespace those classes. Which will make the distinction between us and openvds much more obvious. Either you are inside the classes that abstracts openvds, or you are on the outside where openvds "doesn't" exist

Reply:
To be brutally honest I would say that if you still think we need to disambiguate the axis like this after the introduction of `openvds::DataHandle` and `openvds::Metadata`, then those classes have ultimately failed at their goal of abstraction.

Reply:
This is a very strict perspective, but I agree with it. After removing the `vdsbuffer` there are also no (other) argument to keep the distinction by adding prefixes structs/functions. 😄

I will push a fixup that will lead to an out-of-date commit message. The commit message has to be fixed during squashing.